// @prisma/client/index.js

/* !!! This is code generated by Prisma. Do not edit directly. !!!
/* eslint-disable */
// biome-ignore-all lint: generated file

Object.defineProperty(exports, "__esModule", { value: true });

const {
  PrismaClientKnownRequestError,
  PrismaClientUnknownRequestError,
  PrismaClientRustPanicError,
  PrismaClientInitializationError,
  PrismaClientValidationError,
  getPrismaClient,
  sqltag,
  empty,
  join,
  raw,
  skip,
  Decimal,
  Debug,
  DbNull,
  JsonNull,
  AnyNull,
  NullTypes,
  makeStrictEnum,
  Extensions,
  warnOnce,
  defineDmmfProperty,
  Public,
  getRuntime,
  createParam,
} = require('./runtime/client.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 7.2.0
 * Query Engine version: 0c8ef2ce45c83248ab3df073180d5eda9e8be7a3
 */
Prisma.prismaVersion = {
  client: "7.2.0",
  engine: "0c8ef2ce45c83248ab3df073180d5eda9e8be7a3"
}

Prisma.PrismaClientKnownRequestError = PrismaClientKnownRequestError;
Prisma.PrismaClientUnknownRequestError = PrismaClientUnknownRequestError
Prisma.PrismaClientRustPanicError = PrismaClientRustPanicError
Prisma.PrismaClientInitializationError = PrismaClientInitializationError
Prisma.PrismaClientValidationError = PrismaClientValidationError
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = sqltag
Prisma.empty = empty
Prisma.join = join
Prisma.raw = raw
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = Extensions.getExtensionContext
Prisma.defineExtension = Extensions.defineExtension

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = DbNull
Prisma.JsonNull = JsonNull
Prisma.AnyNull = AnyNull

Prisma.NullTypes = NullTypes




  const path = require('path')

/**
 * Enums
 */
exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.PropertyScalarFieldEnum = {
  id: 'id',
  name: 'name',
  description: 'description',
  pricePerMonth: 'pricePerMonth',
  securityDeposit: 'securityDeposit',
  photoUrls: 'photoUrls',
  amenities: 'amenities',
  highlights: 'highlights',
  closestUniversities: 'closestUniversities',
  accreditedBy: 'accreditedBy',
  closestUniversity: 'closestUniversity',
  closestCampuses: 'closestCampuses',
  isPetsAllowed: 'isPetsAllowed',
  isParkingIncluded: 'isParkingIncluded',
  isNsfassAccredited: 'isNsfassAccredited',
  beds: 'beds',
  baths: 'baths',
  kitchens: 'kitchens',
  squareFeet: 'squareFeet',
  propertyType: 'propertyType',
  postedDate: 'postedDate',
  averageRating: 'averageRating',
  numberOfReviews: 'numberOfReviews',
  locationId: 'locationId',
  managerCognitoId: 'managerCognitoId',
  redirectType: 'redirectType',
  whatsappNumber: 'whatsappNumber',
  customLink: 'customLink'
};

exports.Prisma.RoomScalarFieldEnum = {
  id: 'id',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  name: 'name',
  description: 'description',
  pricePerMonth: 'pricePerMonth',
  securityDeposit: 'securityDeposit',
  topUp: 'topUp',
  squareFeet: 'squareFeet',
  beds: 'beds',
  baths: 'baths',
  photoUrls: 'photoUrls',
  amenities: 'amenities',
  isAvailable: 'isAvailable',
  availableFrom: 'availableFrom',
  roomType: 'roomType',
  capacity: 'capacity',
  features: 'features',
  propertyId: 'propertyId'
};

exports.Prisma.ManagerScalarFieldEnum = {
  id: 'id',
  cognitoId: 'cognitoId',
  name: 'name',
  email: 'email',
  phoneNumber: 'phoneNumber',
  status: 'status',
  approvedAt: 'approvedAt',
  updatedAt: 'updatedAt',
  createdAt: 'createdAt'
};

exports.Prisma.AdminScalarFieldEnum = {
  id: 'id',
  cognitoId: 'cognitoId',
  name: 'name',
  email: 'email',
  phoneNumber: 'phoneNumber',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TenantScalarFieldEnum = {
  id: 'id',
  cognitoId: 'cognitoId',
  name: 'name',
  email: 'email',
  phoneNumber: 'phoneNumber',
  referredBy: 'referredBy'
};

exports.Prisma.LocationScalarFieldEnum = {
  id: 'id',
  address: 'address',
  city: 'city',
  suburb: 'suburb',
  state: 'state',
  country: 'country',
  postalCode: 'postalCode'
};

exports.Prisma.ApplicationScalarFieldEnum = {
  id: 'id',
  applicationDate: 'applicationDate',
  status: 'status',
  propertyId: 'propertyId',
  roomId: 'roomId',
  tenantCognitoId: 'tenantCognitoId',
  name: 'name',
  email: 'email',
  phoneNumber: 'phoneNumber',
  message: 'message',
  leaseId: 'leaseId'
};

exports.Prisma.LeaseScalarFieldEnum = {
  id: 'id',
  startDate: 'startDate',
  endDate: 'endDate',
  rent: 'rent',
  deposit: 'deposit',
  propertyId: 'propertyId',
  tenantCognitoId: 'tenantCognitoId'
};

exports.Prisma.PaymentScalarFieldEnum = {
  id: 'id',
  amountDue: 'amountDue',
  amountPaid: 'amountPaid',
  dueDate: 'dueDate',
  paymentDate: 'paymentDate',
  paymentStatus: 'paymentStatus',
  leaseId: 'leaseId'
};

exports.Prisma.ReviewScalarFieldEnum = {
  id: 'id',
  rating: 'rating',
  comment: 'comment',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  propertyId: 'propertyId',
  tenantCognitoId: 'tenantCognitoId',
  tenantName: 'tenantName'
};

exports.Prisma.ReferralScalarFieldEnum = {
  id: 'id',
  referralCode: 'referralCode',
  referrerCognitoId: 'referrerCognitoId',
  referredCognitoId: 'referredCognitoId',
  referredEmail: 'referredEmail',
  createdAt: 'createdAt',
  completedAt: 'completedAt',
  isCompleted: 'isCompleted',
  voucherGenerated: 'voucherGenerated'
};

exports.Prisma.VoucherScalarFieldEnum = {
  id: 'id',
  code: 'code',
  ownerCognitoId: 'ownerCognitoId',
  discountAmount: 'discountAmount',
  discountPercent: 'discountPercent',
  status: 'status',
  expiresAt: 'expiresAt',
  usedAt: 'usedAt',
  createdAt: 'createdAt',
  referralId: 'referralId'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};
exports.Highlight = exports.$Enums.Highlight = {
  HighSpeedInternetAccess: 'HighSpeedInternetAccess',
  WasherDryer: 'WasherDryer',
  AirConditioning: 'AirConditioning',
  Heating: 'Heating',
  SmokeFree: 'SmokeFree',
  CableReady: 'CableReady',
  SatelliteTV: 'SatelliteTV',
  DoubleVanities: 'DoubleVanities',
  TubShower: 'TubShower',
  Intercom: 'Intercom',
  SprinklerSystem: 'SprinklerSystem',
  RecentlyRenovated: 'RecentlyRenovated',
  CloseToTransit: 'CloseToTransit',
  GreatView: 'GreatView',
  QuietNeighborhood: 'QuietNeighborhood',
  BedroomCleaning: 'BedroomCleaning',
  BackupWater: 'BackupWater',
  BackupElectricity: 'BackupElectricity',
  Refrigerator: 'Refrigerator',
  SwimmingPool: 'SwimmingPool',
  SleepoversAllowed: 'SleepoversAllowed'
};

exports.Amenity = exports.$Enums.Amenity = {
  WasherDryer: 'WasherDryer',
  AirConditioning: 'AirConditioning',
  Dishwasher: 'Dishwasher',
  HighSpeedInternet: 'HighSpeedInternet',
  HardwoodFloors: 'HardwoodFloors',
  WalkInClosets: 'WalkInClosets',
  Microwave: 'Microwave',
  Refrigerator: 'Refrigerator',
  Pool: 'Pool',
  Gym: 'Gym',
  Parking: 'Parking',
  PetsAllowed: 'PetsAllowed',
  WiFi: 'WiFi',
  OnSiteSecurity: 'OnSiteSecurity',
  TransportToCampus: 'TransportToCampus',
  OutdoorChillArea: 'OutdoorChillArea',
  SecurityCameras: 'SecurityCameras',
  Cafe: 'Cafe',
  LaundryFacilities: 'LaundryFacilities',
  GamesRoom: 'GamesRoom',
  TVRoom: 'TVRoom',
  ComputerRoom: 'ComputerRoom',
  StudyLabs: 'StudyLabs',
  Heater: 'Heater',
  Fan: 'Fan',
  Stove: 'Stove',
  Oven: 'Oven',
  BiometricAccess: 'BiometricAccess'
};

exports.PropertyType = exports.$Enums.PropertyType = {
  Rooms: 'Rooms',
  Tinyhouse: 'Tinyhouse',
  Apartment: 'Apartment',
  Villa: 'Villa',
  Townhouse: 'Townhouse',
  Cottage: 'Cottage'
};

exports.University = exports.$Enums.University = {
  UCT: 'UCT',
  WITS: 'WITS',
  UJ: 'UJ',
  UKZN: 'UKZN',
  UWC: 'UWC',
  UP: 'UP',
  SU: 'SU',
  CPUT: 'CPUT',
  TUT: 'TUT',
  UNISA: 'UNISA',
  NWU: 'NWU',
  UFS: 'UFS',
  UFH: 'UFH',
  RU: 'RU',
  WSU: 'WSU',
  UL: 'UL',
  UZ: 'UZ',
  CUT: 'CUT',
  MUT: 'MUT',
  SPU: 'SPU'
};

exports.ApplicationStatus = exports.$Enums.ApplicationStatus = {
  Pending: 'Pending',
  Denied: 'Denied',
  Approved: 'Approved'
};

exports.PaymentStatus = exports.$Enums.PaymentStatus = {
  Pending: 'Pending',
  Paid: 'Paid',
  PartiallyPaid: 'PartiallyPaid',
  Overdue: 'Overdue'
};

exports.RoomType = exports.$Enums.RoomType = {
  PRIVATE: 'PRIVATE',
  SHARED: 'SHARED',
  ENTIRE_UNIT: 'ENTIRE_UNIT'
};

exports.RedirectType = exports.$Enums.RedirectType = {
  NONE: 'NONE',
  WHATSAPP: 'WHATSAPP',
  CUSTOM_LINK: 'CUSTOM_LINK',
  BOTH: 'BOTH'
};

exports.ManagerStatus = exports.$Enums.ManagerStatus = {
  Pending: 'Pending',
  Active: 'Active',
  Disabled: 'Disabled',
  Banned: 'Banned'
};

exports.VoucherStatus = exports.$Enums.VoucherStatus = {
  Active: 'Active',
  Used: 'Used',
  Expired: 'Expired'
};

exports.Prisma.ModelName = {
  Property: 'Property',
  Room: 'Room',
  Manager: 'Manager',
  Admin: 'Admin',
  Tenant: 'Tenant',
  Location: 'Location',
  Application: 'Application',
  Lease: 'Lease',
  Payment: 'Payment',
  Review: 'Review',
  Referral: 'Referral',
  Voucher: 'Voucher'
};
/**
 * Create the Client
 */
const config = {
  "previewFeatures": [
    "postgresqlExtensions"
  ],
  "clientVersion": "7.2.0",
  "engineVersion": "0c8ef2ce45c83248ab3df073180d5eda9e8be7a3",
  "activeProvider": "postgresql",
  "inlineSchema": "generator client {\n  provider        = \"prisma-client-js\"\n  previewFeatures = [\"postgresqlExtensions\"]\n  output          = \"node_modules/@prisma/client\"\n}\n\ndatasource db {\n  provider = \"postgresql\"\n  // Connection URL moved to `prisma.config.cjs` for Prisma v7+ compatibility\n}\n\nenum Highlight {\n  HighSpeedInternetAccess\n  WasherDryer\n  AirConditioning\n  Heating\n  SmokeFree\n  CableReady\n  SatelliteTV\n  DoubleVanities\n  TubShower\n  Intercom\n  SprinklerSystem\n  RecentlyRenovated\n  CloseToTransit\n  GreatView\n  QuietNeighborhood\n  BedroomCleaning\n  BackupWater\n  BackupElectricity\n  Refrigerator\n  SwimmingPool\n  SleepoversAllowed\n}\n\nenum Amenity {\n  WasherDryer\n  AirConditioning\n  Dishwasher\n  HighSpeedInternet\n  HardwoodFloors\n  WalkInClosets\n  Microwave\n  Refrigerator\n  Pool\n  Gym\n  Parking\n  PetsAllowed\n  WiFi\n  OnSiteSecurity\n  TransportToCampus\n  OutdoorChillArea\n  SecurityCameras\n  Cafe\n  LaundryFacilities\n  GamesRoom\n  TVRoom\n  ComputerRoom\n  StudyLabs\n  Heater\n  Fan\n  Stove\n  Oven\n  BiometricAccess\n}\n\nenum PropertyType {\n  Rooms\n  Tinyhouse\n  Apartment\n  Villa\n  Townhouse\n  Cottage\n}\n\nenum University {\n  UCT\n  WITS\n  UJ\n  UKZN\n  UWC\n  UP\n  SU\n  CPUT\n  TUT\n  UNISA\n  NWU\n  UFS\n  UFH\n  RU\n  WSU\n  UL\n  UZ\n  CUT\n  MUT\n  SPU\n}\n\nenum ApplicationStatus {\n  Pending\n  Denied\n  Approved\n}\n\nenum PaymentStatus {\n  Pending\n  Paid\n  PartiallyPaid\n  Overdue\n}\n\nenum RoomType {\n  PRIVATE\n  SHARED\n  ENTIRE_UNIT\n}\n\nenum RedirectType {\n  NONE\n  WHATSAPP\n  CUSTOM_LINK\n  BOTH\n}\n\nmodel Property {\n  id                  Int          @id @default(autoincrement())\n  name                String\n  description         String\n  pricePerMonth       Float\n  securityDeposit     Float\n  photoUrls           String[]\n  amenities           Amenity[]\n  highlights          Highlight[]\n  closestUniversities University[]\n  accreditedBy        University[]\n  closestUniversity   University?\n  closestCampuses     String[]     @default([])\n  isPetsAllowed       Boolean      @default(false)\n  isParkingIncluded   Boolean      @default(false)\n  isNsfassAccredited  Boolean      @default(false)\n  beds                Int\n  baths               Float\n  kitchens            Int          @default(0)\n  squareFeet          Int\n  propertyType        PropertyType\n  postedDate          DateTime     @default(now())\n  averageRating       Float?       @default(0)\n  numberOfReviews     Int?         @default(0)\n  locationId          Int\n  managerCognitoId    String\n\n  // Redirect settings for applications (property-level defaults)\n  redirectType   RedirectType? @default(NONE)\n  whatsappNumber String?\n  customLink     String?\n\n  // Relations\n  location     Location      @relation(fields: [locationId], references: [id])\n  manager      Manager       @relation(fields: [managerCognitoId], references: [cognitoId])\n  leases       Lease[]\n  applications Application[]\n  favoritedBy  Tenant[]      @relation(\"TenantFavorites\")\n  tenants      Tenant[]      @relation(\"TenantProperties\")\n  rooms        Room[]        @relation(\"PropertyRooms\")\n  reviews      Review[]\n}\n\nmodel Room {\n  id              Int       @id @default(autoincrement())\n  createdAt       DateTime  @default(now())\n  updatedAt       DateTime  @updatedAt\n  name            String\n  description     String?   @db.Text\n  pricePerMonth   Float\n  securityDeposit Float     @default(0)\n  topUp           Float     @default(0)\n  squareFeet      Int?\n  beds            Int       @default(1)\n  baths           Float     @default(1)\n  photoUrls       String[]  @default([])\n  amenities       String[]  @default([])\n  isAvailable     Boolean   @default(true)\n  availableFrom   DateTime?\n  roomType        RoomType  @default(PRIVATE)\n  capacity        Int       @default(1)\n  features        String[]  @default([])\n\n  // Relations\n  property     Property      @relation(\"PropertyRooms\", fields: [propertyId], references: [id], onDelete: Cascade)\n  propertyId   Int\n  applications Application[]\n\n  @@index([propertyId])\n}\n\nenum ManagerStatus {\n  Pending\n  Active\n  Disabled\n  Banned\n}\n\nmodel Manager {\n  id          Int           @id @default(autoincrement())\n  cognitoId   String        @unique\n  name        String\n  email       String\n  phoneNumber String\n  status      ManagerStatus @default(Pending)\n  approvedAt  DateTime?\n  updatedAt   DateTime      @updatedAt\n  createdAt   DateTime      @default(now())\n\n  managedProperties Property[]\n}\n\nmodel Admin {\n  id          Int      @id @default(autoincrement())\n  cognitoId   String   @unique\n  name        String\n  email       String\n  phoneNumber String?\n  createdAt   DateTime @default(now())\n  updatedAt   DateTime @updatedAt\n}\n\nmodel Tenant {\n  id          Int     @id @default(autoincrement())\n  cognitoId   String  @unique\n  name        String\n  email       String\n  phoneNumber String\n  referredBy  String? // Referral code used during signup\n\n  properties        Property[]    @relation(\"TenantProperties\")\n  favorites         Property[]    @relation(\"TenantFavorites\")\n  applications      Application[]\n  leases            Lease[]\n  reviews           Review[]\n  referralsMade     Referral[]    @relation(\"ReferrerRelation\")\n  referralsReceived Referral[]    @relation(\"ReferredRelation\")\n  vouchers          Voucher[]\n}\n\nmodel Location {\n  id          Int                                   @id @default(autoincrement())\n  address     String\n  city        String\n  suburb      String?\n  state       String\n  country     String\n  postalCode  String\n  coordinates Unsupported(\"geography(Point, 4326)\")\n\n  properties Property[]\n}\n\nmodel Application {\n  id              Int               @id @default(autoincrement())\n  applicationDate DateTime\n  status          ApplicationStatus\n  propertyId      Int\n  roomId          Int?\n  tenantCognitoId String?\n  name            String\n  email           String\n  phoneNumber     String\n  message         String?\n  leaseId         Int?              @unique\n\n  property Property @relation(fields: [propertyId], references: [id])\n  room     Room?    @relation(fields: [roomId], references: [id])\n  tenant   Tenant?  @relation(fields: [tenantCognitoId], references: [cognitoId])\n  lease    Lease?   @relation(fields: [leaseId], references: [id])\n}\n\nmodel Lease {\n  id              Int      @id @default(autoincrement())\n  startDate       DateTime\n  endDate         DateTime\n  rent            Float\n  deposit         Float\n  propertyId      Int\n  tenantCognitoId String\n\n  property    Property     @relation(fields: [propertyId], references: [id])\n  tenant      Tenant       @relation(fields: [tenantCognitoId], references: [cognitoId])\n  application Application?\n  payments    Payment[]\n}\n\nmodel Payment {\n  id            Int           @id @default(autoincrement())\n  amountDue     Float\n  amountPaid    Float\n  dueDate       DateTime\n  paymentDate   DateTime\n  paymentStatus PaymentStatus\n  leaseId       Int\n\n  lease Lease @relation(fields: [leaseId], references: [id])\n}\n\nmodel Review {\n  id              Int      @id @default(autoincrement())\n  rating          Int // 1-5 stars\n  comment         String?  @db.Text\n  createdAt       DateTime @default(now())\n  updatedAt       DateTime @updatedAt\n  propertyId      Int\n  tenantCognitoId String\n  tenantName      String? // Store tenant name for display\n\n  property Property @relation(fields: [propertyId], references: [id], onDelete: Cascade)\n  tenant   Tenant   @relation(fields: [tenantCognitoId], references: [cognitoId], onDelete: Cascade)\n\n  // Prevent duplicate reviews per tenant per property\n  @@unique([propertyId, tenantCognitoId])\n  @@index([propertyId])\n  @@index([tenantCognitoId])\n}\n\nenum VoucherStatus {\n  Active\n  Used\n  Expired\n}\n\nmodel Referral {\n  id                Int       @id @default(autoincrement())\n  referralCode      String    @unique\n  referrerCognitoId String // The person who refers others\n  referredCognitoId String? // The person who was referred (null if not signed up yet)\n  referredEmail     String? // Email of the referred person (for tracking before signup)\n  createdAt         DateTime  @default(now())\n  completedAt       DateTime? // When the referred person completed a qualifying action\n  isCompleted       Boolean   @default(false)\n  voucherGenerated  Boolean   @default(false)\n\n  // Relations\n  referrer Tenant    @relation(\"ReferrerRelation\", fields: [referrerCognitoId], references: [cognitoId], onDelete: Cascade)\n  referred Tenant?   @relation(\"ReferredRelation\", fields: [referredCognitoId], references: [cognitoId], onDelete: Cascade)\n  vouchers Voucher[]\n\n  @@index([referralCode])\n  @@index([referrerCognitoId])\n  @@index([referredCognitoId])\n}\n\nmodel Voucher {\n  id              Int           @id @default(autoincrement())\n  code            String        @unique\n  ownerCognitoId  String\n  discountAmount  Float // Discount in currency\n  discountPercent Float? // Optional percentage discount\n  status          VoucherStatus @default(Active)\n  expiresAt       DateTime\n  usedAt          DateTime?\n  createdAt       DateTime      @default(now())\n  referralId      Int? // Link to the referral that generated this voucher\n\n  // Relations\n  owner    Tenant    @relation(fields: [ownerCognitoId], references: [cognitoId], onDelete: Cascade)\n  referral Referral? @relation(fields: [referralId], references: [id])\n\n  @@index([code])\n  @@index([ownerCognitoId])\n  @@index([status])\n}\n"
}

config.runtimeDataModel = JSON.parse("{\"models\":{\"Property\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"description\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"pricePerMonth\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"securityDeposit\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"photoUrls\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"amenities\",\"kind\":\"enum\",\"type\":\"Amenity\"},{\"name\":\"highlights\",\"kind\":\"enum\",\"type\":\"Highlight\"},{\"name\":\"closestUniversities\",\"kind\":\"enum\",\"type\":\"University\"},{\"name\":\"accreditedBy\",\"kind\":\"enum\",\"type\":\"University\"},{\"name\":\"closestUniversity\",\"kind\":\"enum\",\"type\":\"University\"},{\"name\":\"closestCampuses\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"isPetsAllowed\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"isParkingIncluded\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"isNsfassAccredited\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"beds\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"baths\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"kitchens\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"squareFeet\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"propertyType\",\"kind\":\"enum\",\"type\":\"PropertyType\"},{\"name\":\"postedDate\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"averageRating\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"numberOfReviews\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"locationId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"managerCognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"redirectType\",\"kind\":\"enum\",\"type\":\"RedirectType\"},{\"name\":\"whatsappNumber\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"customLink\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"location\",\"kind\":\"object\",\"type\":\"Location\",\"relationName\":\"LocationToProperty\"},{\"name\":\"manager\",\"kind\":\"object\",\"type\":\"Manager\",\"relationName\":\"ManagerToProperty\"},{\"name\":\"leases\",\"kind\":\"object\",\"type\":\"Lease\",\"relationName\":\"LeaseToProperty\"},{\"name\":\"applications\",\"kind\":\"object\",\"type\":\"Application\",\"relationName\":\"ApplicationToProperty\"},{\"name\":\"favoritedBy\",\"kind\":\"object\",\"type\":\"Tenant\",\"relationName\":\"TenantFavorites\"},{\"name\":\"tenants\",\"kind\":\"object\",\"type\":\"Tenant\",\"relationName\":\"TenantProperties\"},{\"name\":\"rooms\",\"kind\":\"object\",\"type\":\"Room\",\"relationName\":\"PropertyRooms\"},{\"name\":\"reviews\",\"kind\":\"object\",\"type\":\"Review\",\"relationName\":\"PropertyToReview\"}],\"dbName\":null},\"Room\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"description\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"pricePerMonth\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"securityDeposit\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"topUp\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"squareFeet\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"beds\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"baths\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"photoUrls\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"amenities\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"isAvailable\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"availableFrom\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"roomType\",\"kind\":\"enum\",\"type\":\"RoomType\"},{\"name\":\"capacity\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"features\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"property\",\"kind\":\"object\",\"type\":\"Property\",\"relationName\":\"PropertyRooms\"},{\"name\":\"propertyId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"applications\",\"kind\":\"object\",\"type\":\"Application\",\"relationName\":\"ApplicationToRoom\"}],\"dbName\":null},\"Manager\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"cognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"email\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"phoneNumber\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"status\",\"kind\":\"enum\",\"type\":\"ManagerStatus\"},{\"name\":\"approvedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"managedProperties\",\"kind\":\"object\",\"type\":\"Property\",\"relationName\":\"ManagerToProperty\"}],\"dbName\":null},\"Admin\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"cognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"email\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"phoneNumber\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null},\"Tenant\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"cognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"email\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"phoneNumber\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"referredBy\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"properties\",\"kind\":\"object\",\"type\":\"Property\",\"relationName\":\"TenantProperties\"},{\"name\":\"favorites\",\"kind\":\"object\",\"type\":\"Property\",\"relationName\":\"TenantFavorites\"},{\"name\":\"applications\",\"kind\":\"object\",\"type\":\"Application\",\"relationName\":\"ApplicationToTenant\"},{\"name\":\"leases\",\"kind\":\"object\",\"type\":\"Lease\",\"relationName\":\"LeaseToTenant\"},{\"name\":\"reviews\",\"kind\":\"object\",\"type\":\"Review\",\"relationName\":\"ReviewToTenant\"},{\"name\":\"referralsMade\",\"kind\":\"object\",\"type\":\"Referral\",\"relationName\":\"ReferrerRelation\"},{\"name\":\"referralsReceived\",\"kind\":\"object\",\"type\":\"Referral\",\"relationName\":\"ReferredRelation\"},{\"name\":\"vouchers\",\"kind\":\"object\",\"type\":\"Voucher\",\"relationName\":\"TenantToVoucher\"}],\"dbName\":null},\"Location\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"address\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"city\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"suburb\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"state\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"country\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"postalCode\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"properties\",\"kind\":\"object\",\"type\":\"Property\",\"relationName\":\"LocationToProperty\"}],\"dbName\":null},\"Application\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"applicationDate\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"status\",\"kind\":\"enum\",\"type\":\"ApplicationStatus\"},{\"name\":\"propertyId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"roomId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"tenantCognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"email\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"phoneNumber\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"message\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"leaseId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"property\",\"kind\":\"object\",\"type\":\"Property\",\"relationName\":\"ApplicationToProperty\"},{\"name\":\"room\",\"kind\":\"object\",\"type\":\"Room\",\"relationName\":\"ApplicationToRoom\"},{\"name\":\"tenant\",\"kind\":\"object\",\"type\":\"Tenant\",\"relationName\":\"ApplicationToTenant\"},{\"name\":\"lease\",\"kind\":\"object\",\"type\":\"Lease\",\"relationName\":\"ApplicationToLease\"}],\"dbName\":null},\"Lease\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"startDate\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"endDate\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"rent\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"deposit\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"propertyId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"tenantCognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"property\",\"kind\":\"object\",\"type\":\"Property\",\"relationName\":\"LeaseToProperty\"},{\"name\":\"tenant\",\"kind\":\"object\",\"type\":\"Tenant\",\"relationName\":\"LeaseToTenant\"},{\"name\":\"application\",\"kind\":\"object\",\"type\":\"Application\",\"relationName\":\"ApplicationToLease\"},{\"name\":\"payments\",\"kind\":\"object\",\"type\":\"Payment\",\"relationName\":\"LeaseToPayment\"}],\"dbName\":null},\"Payment\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"amountDue\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"amountPaid\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"dueDate\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"paymentDate\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"paymentStatus\",\"kind\":\"enum\",\"type\":\"PaymentStatus\"},{\"name\":\"leaseId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"lease\",\"kind\":\"object\",\"type\":\"Lease\",\"relationName\":\"LeaseToPayment\"}],\"dbName\":null},\"Review\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"rating\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"comment\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"propertyId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"tenantCognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"tenantName\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"property\",\"kind\":\"object\",\"type\":\"Property\",\"relationName\":\"PropertyToReview\"},{\"name\":\"tenant\",\"kind\":\"object\",\"type\":\"Tenant\",\"relationName\":\"ReviewToTenant\"}],\"dbName\":null},\"Referral\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"referralCode\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"referrerCognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"referredCognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"referredEmail\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"completedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"isCompleted\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"voucherGenerated\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"referrer\",\"kind\":\"object\",\"type\":\"Tenant\",\"relationName\":\"ReferrerRelation\"},{\"name\":\"referred\",\"kind\":\"object\",\"type\":\"Tenant\",\"relationName\":\"ReferredRelation\"},{\"name\":\"vouchers\",\"kind\":\"object\",\"type\":\"Voucher\",\"relationName\":\"ReferralToVoucher\"}],\"dbName\":null},\"Voucher\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"code\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"ownerCognitoId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"discountAmount\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"discountPercent\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"status\",\"kind\":\"enum\",\"type\":\"VoucherStatus\"},{\"name\":\"expiresAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"usedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"referralId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"owner\",\"kind\":\"object\",\"type\":\"Tenant\",\"relationName\":\"TenantToVoucher\"},{\"name\":\"referral\",\"kind\":\"object\",\"type\":\"Referral\",\"relationName\":\"ReferralToVoucher\"}],\"dbName\":null}},\"enums\":{},\"types\":{}}")
defineDmmfProperty(exports.Prisma, config.runtimeDataModel)
config.compilerWasm = {
      getRuntime: async () => require('./query_compiler_bg.js'),
      getQueryCompilerWasmModule: async () => {
        const { Buffer } = require('node:buffer')
        const { wasm } = require('./query_compiler_bg.wasm-base64.js')
        const queryCompilerWasmFileBytes = Buffer.from(wasm, 'base64')

        return new WebAssembly.Module(queryCompilerWasmFileBytes)
      }
    }

const PrismaClient = getPrismaClient(config)
exports.PrismaClient = PrismaClient
Object.assign(exports, Prisma)
